diff --git a/sys/conf/kern.mk b/sys/conf/kern.mk
index 5d07a1d9ad3..5279e80fd19 100644
--- a/sys/conf/kern.mk
+++ b/sys/conf/kern.mk
@@ -142,10 +142,10 @@ CFLAGS += -mgeneral-regs-only
 # Reserve x18 for pcpu data
 CFLAGS += -ffixed-x18
 # Build with BTI+PAC
-CFLAGS += -mbranch-protection=standard
-.if ${LINKER_FEATURES:Mbti-report}
-LDFLAGS += -Wl,-zbti-report=error
-.endif
+#CFLAGS += -mbranch-protection=standard
+#.if ${LINKER_FEATURES:Mbti-report}
+#LDFLAGS += -Wl,-zbti-report=error
+#.endif
 # TODO: support outline atomics
 CFLAGS += -mno-outline-atomics
 INLINE_LIMIT?=	8000
