commit ee5bc49510da5ac65953bfba58f0046fda8cedb0
Author: Ayrton Munoz <a.munoz3327@gmail.com>
Date:   Mon Nov 10 23:25:51 2025 -0500

    conf: Integrate rust into kernel build system

diff --git a/sys/conf/kern.post.mk b/sys/conf/kern.post.mk
index 2abf1889d33..68c1c697d8c 100644
--- a/sys/conf/kern.post.mk
+++ b/sys/conf/kern.post.mk
@@ -7,6 +7,7 @@
 # should be defined in the kern.pre.mk so that port makefiles can
 # override or augment them.
 
+.include "../rust/kern.rustroot.mk"
 .if defined(DTS) || defined(DTSO) || defined(FDT_DTS_FILE)
 .include "dtb.build.mk"
 
diff --git a/sys/conf/kern.pre.mk b/sys/conf/kern.pre.mk
index d1556660094..8ffcb07b929 100644
--- a/sys/conf/kern.pre.mk
+++ b/sys/conf/kern.pre.mk
@@ -21,6 +21,7 @@ _srcconf_included_:
 .include <bsd.own.mk>
 .include <bsd.compiler.mk>
 .include "kern.opts.mk"
+.include "../rust/kern.rust.mk"
 .-include <local.kern.pre.mk>
 
 # The kernel build always occurs in the object directory which is .CURDIR.
@@ -293,8 +294,8 @@ IPFILTER_C=		${NORMAL_C} ${IPFILTER_CFLAGS}
 
 GEN_CFILES= $S/$M/$M/genassym.c ${MFILES:T:S/.m$/.c/}
 SYSTEM_CFILES= config.c env.c hints.c vnode_if.c
-SYSTEM_DEP= Makefile ${SYSTEM_OBJS}
-SYSTEM_OBJS= locore.o ${MDOBJS} ${OBJS}
+SYSTEM_DEP= Makefile ${SYSTEM_OBJS} ${RUST_MAKEFILE}
+SYSTEM_OBJS= locore.o ${MDOBJS} ${OBJS} ${RUST_OBJS}
 SYSTEM_OBJS+= ${SYSTEM_CFILES:.c=.o}
 SYSTEM_OBJS+= force-dynamic-hack.pico
 
diff --git a/sys/conf/kmod.mk b/sys/conf/kmod.mk
index c35116dfcaa..8c991e91f30 100644
--- a/sys/conf/kmod.mk
+++ b/sys/conf/kmod.mk
@@ -226,6 +226,11 @@ SRCS+=${SRCS.${_o}}
 
 OBJS+=	${SRCS:N*.h:R:S/$/.o/g}
 
+.if defined(RUST_SRCS) && !empty(RUST_SRCS)
+.include "${SYSDIR}/rust/kmod.rust.mk"
+OBJS+=	${RUST_OBJS}
+.endif
+
 .if !defined(PROG)
 PROG=	${KMOD}.ko
 .endif
