diff --git a/sys/conf/kern.post.mk b/sys/conf/kern.post.mk
index 7cdfd17778d..b6cb6c3c0cf 100644
--- a/sys/conf/kern.post.mk
+++ b/sys/conf/kern.post.mk
@@ -7,6 +7,8 @@
 # should be defined in the kern.pre.mk so that port makefiles can
 # override or augment them.
 
+.include "../rust/kern.rust.mk"
+
 .if defined(DTS) || defined(DTSO) || defined(FDT_DTS_FILE)
 .include "dtb.build.mk"
 
diff --git a/sys/conf/kern.pre.mk b/sys/conf/kern.pre.mk
index 0251486247d..9ea1dca4789 100644
--- a/sys/conf/kern.pre.mk
+++ b/sys/conf/kern.pre.mk
@@ -302,7 +302,7 @@ IPFILTER_C=		${NORMAL_C} ${IPFILTER_CFLAGS}
 GEN_CFILES= $S/$M/$M/genassym.c ${MFILES:T:S/.m$/.c/}
 SYSTEM_CFILES= config.c env.c hints.c vnode_if.c
 SYSTEM_DEP= Makefile ${SYSTEM_OBJS}
-SYSTEM_OBJS= locore.o ${MDOBJS} ${OBJS}
+SYSTEM_OBJS= locore.o ${MDOBJS} ${OBJS} ${RUST_OBJS}
 SYSTEM_OBJS+= ${SYSTEM_CFILES:.c=.o}
 SYSTEM_OBJS+= force-dynamic-hack.pico
 
